packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -13,7 +13,11 @@ import { isArray } from 'ember-runtime/utils';
 
 function reduceMacro(dependentKey, callback, initialValue) {
   return computed(`${dependentKey}.[]`, function() {
-    return get(this, dependentKey).reduce((previousValue, currentValue, index, array) => {
+    const arr = get(this, dependentKey);
+
+    if (!isArray(arr)) { return initialValue; }
```

Comment:
I believe this check should actually be

`(arr === null || arr === undefined)`

If `arr` is an object and doesn't have reduce we should throw, but not if it is just not an array or an array like.